src/main/java/kostka/moviecatalog/service/communication/RestCommunicationService.java

```diff
@@ -0,0 +1,25 @@
+package kostka.moviecatalog.service.communication;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service("RestCommunicationService")
```

Comment:
If you want to switch implementations easily in the future... Instead of a name, I'd vote to use Spring Boot's @ConditionalXYZ annotations to selectively register beans.

And maybe for simplicity use some property in application.properties for configuration.

Reply:
thanks for the hint, I would create a separate story for Conditional beans, what do you think?

Reply:
#178